kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/channel/Channel.java

```diff
@@ -73,17 +72,10 @@ public Channel(final String name, final ChannelType type, final DataType valueTy
         requireNonNull(valueType, "Channel value type cannot be null");
         requireNonNull(config, "Channel configuration cannot be null");
 
-        if (!Channel.isValidChannelName(name)) {
```

Comment:
I've removed it because I think that the channel name validation is more related to the asset and to the way it stores the channel configuration, for this reason I moved all the channel name validation stuff to the asset package.
For example if you are using a Driver directly, without using an asset, there is no need for validating the channel name